src/main/java/de/teamlapen/vampirism/entity/ai/EntityAIUseAction.java

```diff
@@ -0,0 +1,149 @@
+ * can be used as AITask in custom task list or with {@link handle()} in UpdateLiving in an Entity
+ */
+public class EntityAIUseAction<T extends EntityVampirism & IFactionEntity & IAdjustableLevel> extends EntityAIBase {
+
+    private T entity;
+    private List<IEntityAction> availableActions;
+    private int cooldown = 0;
+    private int duration = 0;
+    private IEntityAction action;
+    private Random rand = new Random();
+    private boolean flag;
+
+    public EntityAIUseAction(T entityIn, List<IEntityAction> actions) {
+        this.entity = entityIn;
+        this.availableActions = actions;
+    }
+
+    @Override
+    public boolean shouldExecute() {
+        if (entity.getAttackTarget() instanceof EntityPlayer && !availableActions.isEmpty()) {
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public void startExecuting() {
+        action = null;
+        cooldown = 100;
+        duration = -1;
+    }
+
+    /**
+     * Keep ticking a continuous task that has already been started,
+     * 
+     * {@link duration} > 0 action in progress,
+     * {@link duration} = 0 action will be deactivated,
+     * {@link duration} = -1 no action is active,
+     * {@link duration} <= 0 && {@link cooldown} > 0 actions on cooldown,
+     * {@link duration} <= 0 && {@link cooldown} = 0 new action will be set & activated
+     */
+    @Override
+    public void updateTask() {
+        if (duration > 0) {
+            duration--;
+            updateAction();
+        } else if (cooldown > 0) {
+            cooldown--;
+        } else {
+            newAction();
+            cooldown = action.getCooldown(entity.getLevel());
+            if (action instanceof ILastingAction) {
+                duration = ((ILastingAction<T>) action).getDuration(entity.getLevel());
+                ((ILastingAction<T>) action).activate(entity);
+            } else if (action instanceof IInstantAction) {
+                ((IInstantAction<T>) action).activate(entity);
+            }
+        }
+
+        /* calls deactivate() for {@link ILastingAction} once per action */
+        if (duration == 0) {
+            if (action instanceof ILastingAction) {
+                ((ILastingAction<T>) action).deactivate(entity);
+            }
+            duration--;
+        }
+    }
+
+    @Override
+    public boolean isInterruptible() {
+        return false;
+    }
+
+    @Override
+    public void resetTask() {
+        if (action instanceof ILastingAction) {
+            ((ILastingAction<T>) action).deactivate(entity);
+        }
+    }
+
+    /**
+     * updates action based on actiontype
+     */
+    private void updateAction() {
+        if (action instanceof ILastingAction) {
+            ((ILastingAction<T>) action).onUpdate(entity, duration);
+        }
+    }
+
+    /**
+     * sets a new random action from {@link availableActions} for the entity,
+     * otherwise a placeholder
+     */
+    private void newAction() {
```

Comment:
I think it would be cleaner if this would just return an action which would then be set as active.
Maybe `selectNewAction` or `getRandomAction` would be a more meaningful name then.

Also, the dummy action seems unessesarry, maybe just return null here, and then check in updateTasks for null (and set a cooldown then)